src/gentropy/datasource/gnomad/ld.py

```diff
@@ -448,3 +449,80 @@ def get_ld_matrix_slice(
                 .alias("r"),
             )
         )
+
+    @staticmethod
+    def get_locus_index(
+        session: Session,
+        study_locus_row: DataFrame,
+        ld_index_path: str,
+        window_size: int = 1_000_000,
+        major_population: str = "nfe",
+    ) -> DataFrame:
+        """Extract hail matrix index from StudyLocus rows.
+
+        Args:
+            session (Session): Spark session
+            study_locus_row (DataFrame): Study-locus row
+            ld_index_path (str): Path to the hail LD index parquet
+            window_size (int): Window size to extract from gnomad matrix
+            major_population (str): Major population to extract from gnomad matrix, default is "nfe"
+
+        Returns:
+            DataFrame: Returns the index of the gnomad matrix for the locus
+
+        """
+        _df = (
+            study_locus_row.withColumn("start", f.col("position") - (window_size / 2))
+            .withColumn("end", f.col("position") + (window_size / 2))
+            .alias("_df")
+        )
+
+        _matrix_index = session.spark.read.parquet(
+            ld_index_path.format(POP=major_population)
+        )
+
+        _index_joined = (
+            _df.alias("df")
+            .join(
+                _matrix_index.alias("matrix_index"),
+                (f.col("df.chromosome") == f.col("matrix_index.chromosome"))
+                & (f.col("df.start") <= f.col("matrix_index.position"))
+                & (f.col("df.end") >= f.col("matrix_index.position")),
+            )
```

Comment:
If all the columns for the resulting table can be selected from the matrix_index table, there's no need to join, it can be a filter. Even if the alleles are not in the columns, we can say:

```python
(
    _matrix_index.
    .filter(
        (f.col("chromosome") == "chr") &
        (f.col("df.position") >= position - (window_size / 2)) &
        (f.col("df.position") <= position + (window_size / 2))
    )
    .select(
        "chromosome",
        "position",
        f.split(f.col("variantId"), "_")[2].alias("referenceAllele"),
        f.split(f.col("variantId"), "_")[3].alias("alternateAllele"),
    )
)
```